.storybook/src/zoom.stories.ts

```diff
@@ -0,0 +1,285 @@
+import * as d3 from 'd3';
+import CanvasLayer from '../../src/layers/CanvasLayer';
+import {
+  OnMountEvent,
+  OnUpdateEvent,
+  OnRescaleEvent,
+  LayerOptions,
+} from '../../src/interfaces';
+import GridLayer from '../../src/layers/GridLayer';
+import ZoomPanHandler from '../../src/control/ZoomPanHandler';
+import { scaleLinear, curveStep } from 'd3';
+
+const width = 600;
+const height = 400;
+
+function createButton(label: string, cb: any){
+  const btn = document.createElement('button');
+  btn.innerHTML = label;
+  btn.setAttribute('style', 'width: 100px;height:32px;margin-top:12px;')
+  btn.onclick = cb;
+  return btn;
+}
+
+class TestLayer extends CanvasLayer {
+  constructor(id: String, options: LayerOptions) {
+    super(id, options);
+
+    this.render = this.render.bind(this);
+  }
+
+  onRescale(event: OnRescaleEvent) {
+    super.onRescale(event);
+
+    this.render(event);
+  }
+
+  onUpdate(event: OnUpdateEvent) {
+    super.onUpdate(event);
+
+    this.render(event);
+  }
+
+  render(event: OnRescaleEvent | OnUpdateEvent) {
```

Comment:
Should be refactored into smaller methods

Reply:
I cleaned this up and shortened it a bit, but given that this is a (slightly silly) example which is not intended to be reused or in need of much maintenance, I don't see the need. The whole TestLayer should not be in this file if we were to adder to best practices, but for a storybook example I think it makes it easier to have all the code in one place and to read the drawing code sequentially.